cfpb/cfgov-refresh
# Update Django-Flags to latest release, use the Flags Debug Toolbar panels, and update Wagtail-Flags

This PR does three things:

- Updates Django-Flags to the latest release, [which includes debug, logging, and state-checking related helpers](https://cfpb.github.io/django-flags/releasenotes/#421).
- Adds the new [Django-Flags Debug Toolbar panels](https://cfpb.github.io/django-flags/debugging/#django-debug-toolbar-panels) to Django Debug Toolbar (and, by necessity, sets the default panels explicitly)
- Updates Wagtail-Flags to the latest *beta* release to include the [admin overhaul](https://github.com/cfpb/wagtail-flags/pull/35). See below about testing the beta.
 
## Testing

1. Check out this branch (`bump-flags`)
2. [Rebuild your python Docker container](https://cfpb.github.io/cfgov-refresh/running-docker/#update-python-dependencies) to get the updated libraries (`docker-compose build python2`)
3. Run Docker (`docker-compose up`)

Once you've got the libraries updated, you can test the Wagtail-Flags beta release's admin overhaul and decide whether you prefer to keep the big-list-of-all-flags-and-conditions that we have no (with added features) or whether you prefer a drill-down approach into flags to see conditions. See https://github.com/cfpb/wagtail-flags/pull/35 for discussion and context.

To try the new drill-down conditions:

1. Visit http://localhost:8000/admin/flags/
2. Browse flags, clicking on the flag names to see conditions

To try one-big-list:

1. Visit http://localhost:8000/admin/flags/
2. Click "Add flag" in the top right
3. Add the flag named `WAGTAILFLAGS_ADMIN_BIG_LIST` and click "Save flag"
4. Visit the `WAGTAILFLAGS_ADMIN_BIG_LIST` flag and click "
Enable WAGTAILFLAGS_ADMIN_BIG_LIST for all requests"
5. View the great big list of all flags and conditions

The next release of Wagtail-Flags is on hold depending on whether the drill-down of flags or the great big list of flags and conditions is preferred. Once that decision is made, we'll strip the feature flag out of Wagtail-Flags, release 4.1.0 proper, and update this PR and merge it.

## Checklist
